SConstruct

```diff
@@ -733,6 +741,31 @@ def should_prepare_targets(style, toolchain, variant):
         if should_prepare_target(t, style, toolchain, variant):
             return True
 
+def should_build_ninja(style, toolchain, variant):
+    """
+    Return True if a ninja build file should be generated.
+
+    Typically, scons will be called as follows to generate a ninja build file:
+    `scons ninja=1 gcc.debug` where `gcc.debug` may be replaced with any of our
+    non-visual studio targets. Raise an exception if we cannot generate the
+    requested ninja build file (for example, if multiple targets are requested).
+    """
+    if 'ninja' not in ARGUMENTS or ARGUMENTS['ninja'] == '0':
+        return False
+    if len(COMMAND_LINE_TARGETS) != 1:
+        raise Exception('Can only generate a ninja file for a single target')
+    cl_target = COMMAND_LINE_TARGETS[0]
+    if 'vcxproj' in cl_target:
+        raise Exception('Cannot generate a ninja file for a vcxproj')
+    s = cl_target.split('.')
+    if ( style == 'unity' and 'nounity' in s or
```

Comment:
@rec The intent of these lines is to filter out targets from the loop that iterates through the styles, toolchains, and variants. Sometimes that loop executes for targets that will not be built. I don't recall the exact configuration, but it was something like `scons.debug` building environments for both unity and nonunity. Since I am limited to building a single `build.ninja` file and I need to make sure I build the right one.